.github/workflows/format.yml

```diff
@@ -0,0 +1,24 @@
+name: Format
+on: [push]
+
+
+jobs:
+  yarn-install:
+    runs-on: ubuntu-latest
+    name: Foundry Tests
+    steps:
+      - uses: actions/checkout@v2     # foundry docs say v3 but npm install uses v2
+        with:
+          submodules: recursive       # ensure forge-std installed
+      - uses: bahmutov/npm-install@v1
+        with:
+          useLockFile: true
+        env:
+          NODE_ENV: development       # install all test tooling 
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      - name: format
+        run: forge fmt --root ./contracts
```

Comment:
Swear i added comments to this already. 
Can u make it so that this creates a commit if there are changes in formatting.

Reply:
This is ready since forge fmt is in a diiferent PR